aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -710,6 +742,32 @@ impl VMChangeSet {
         Ok(())
     }
 
+    fn squash_additional_group_reads_needing_exchange<K>(
+        group_reads_needing_exchange: &mut BTreeMap<StateKey, (WriteOp, u64)>,
+        additional_group_reads_needing_exchange: BTreeMap<StateKey, (WriteOp, u64)>,
+        skip: &BTreeMap<StateKey, K>,
+    ) -> anyhow::Result<(), VMStatus> {
+        for key in skip.keys() {
+            group_reads_needing_exchange.remove(key);
+        }
+        for (key, additional_value) in additional_group_reads_needing_exchange.into_iter() {
+            if skip.contains_key(&key) {
+                continue;
```

Comment:
why not remove from both here? and if we do that, then at least this implementation could actually work with Iterator? @igor-aptos

Reply:
Actually you can pass `mut additional_group_reads_needing_exchange` here because we pass the value, not the reference?